# **Data Science Salaries on US Marketplace**


## Objectives

* Analysing salary trends and how they vary across year, title, experience and location

## Inputs

* The input is "salaries.csv"

## Outputs

* The output is "salaries_processed.csv" and is saved into dataset folder/processed in our project ready for data visualisation.


---

# Change working directory

* We are assuming you will store the notebooks in a subfolder, therefore when running the notebook in the editor, you will need to change the working directory

We need to change the working directory from its current folder to its parent folder
* We access the current directory with os.getcwd()

In [2]:
import os
current_dir = os.getcwd()
current_dir

'd:\\VS-projects\\my_first_project\\Project_1\\jupyter_notebooks'

We want to make the parent of the current directory the new current directory
* os.path.dirname() gets the parent directory
* os.chir() defines the new current directory

In [3]:
os.chdir(os.path.dirname(current_dir))
print("Set a new current directory")

Set a new current directory


Confirm the new current directory

In [4]:
current_dir = os.getcwd()
current_dir

'd:\\VS-projects\\my_first_project\\Project_1'

# Content

* Extraction: loading data from a CSV file

In [5]:
import pandas as pd
df = pd.read_csv('dataset/raw/salaries.csv')
df.head()

work_year experience_level employment_type       job_title  salary  \
0       2025               EX              FT    Head of Data  348516   
1       2025               EX              FT    Head of Data  232344   
2       2025               SE              FT  Data Scientist  145400   
3       2025               SE              FT  Data Scientist   81600   
4       2025               MI              FT        Engineer  160000   

  salary_currency  salary_in_usd employee_residence  remote_ratio  \
0             USD         348516                 US             0   
1             USD         232344                 US             0   
2             USD         145400                 US             0   
3             USD          81600                 US             0   
4             USD         160000                 US           100   

  company_location company_size  
0               US            M  
1               US            M  
2               US            M  
3               US            M  
4               US            M

* Dataset overview

In [ ]:
print(f"DataSet shape: {df.shape}")
print(f"DataSet columns: {df.columns.tolist()}")
print(f"DataSet info:")
df.info()

DataSet shape: (151445, 11)
DataSet columns: ['work_year', 'experience_level', 'employment_type', 'job_title', 'salary', 'salary_currency', 'salary_in_usd', 'employee_residence', 'remote_ratio', 'company_location', 'company_size']
DataSet info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 151445 entries, 0 to 151444
Data columns (total 11 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   work_year           151445 non-null  int64 
 1   experience_level    151445 non-null  object
 2   employment_type     151445 non-null  object
 3   job_title           151445 non-null  object
 4   salary              151445 non-null  int64 
 5   salary_currency     151445 non-null  object
 6   salary_in_usd       151445 non-null  int64 
 7   employee_residence  151445 non-null  object
 8   remote_ratio        151445 non-null  int64 
 9   company_location    151445 non-null  object
 10  company_size        151445 non-null  object
dtypes:

We can extract some valuable information from our dataset by analyzing the above output. Our dataframe contains eleven columns and 151445 rows. Four columns describe variables which represent numbers such as salary and year, and seven variables describe categorical data such as job title and location.

* Statistical overview

In [ ]:
print(f"DataSet description:")
print(df.describe())

DataSet description:
           work_year        salary  salary_in_usd   remote_ratio
count  151445.000000  1.514450e+05  151445.000000  151445.000000
mean     2024.435313  1.628380e+05  157527.458411      20.938625
std         0.671842  2.080124e+05   74150.772377      40.620393
min      2020.000000  1.400000e+04   15000.000000       0.000000
25%      2024.000000  1.060000e+05  105800.000000       0.000000
50%      2025.000000  1.470000e+05  146100.000000       0.000000
75%      2025.000000  1.990000e+05  198000.000000       0.000000
max      2025.000000  3.040000e+07  800000.000000     100.000000


Analysing the DataSet description we can observe that entry-level roles start at 15.000 USD and rise up to 800.000 USD, while the majority of the roles show a salary of 15.7527 USD probably caused by the type of employment and experience level. 


**Data transformation**   
  * Cleaning data
     * Checking for missing values

In [ ]:
missing_values = df.isnull().sum()
missing_columns = df.columns[df.isnull().any()].tolist()
print("Missing values in each column:")
print(missing_values[missing_values > 0])

Missing values in each column:
Series([], dtype: int64)


No missing data were found in the dataset.

* Checking for duplicates

In [ ]:

df.duplicated().any
df.shape


(151445, 11)

No duplicate rows were found in the dataset.

* Checking if variables are in the correct format.

In [ ]:
df.dtypes

work_year              int64
experience_level      object
employment_type       object
job_title             object
salary                 int64
salary_currency       object
salary_in_usd          int64
employee_residence    object
remote_ratio           int64
company_location      object
company_size          object
dtype: object

As we can notice from the above output, all the variables are of the expected data types.

* Dropping no needed columns on our DataFrame
  For a better analysis, we will drop the following columns: employee residence, company size, salary, salary currency, and employment type.

In [ ]:
df.drop(["employee_residence", "company_size", "salary_currency", "salary", "employment_type"], axis=1, inplace=True)

* To make sure our DataFrame has dropped no needed columns we can check the remaining columns.

In [ ]:
print(df.columns.tolist())

['work_year', 'experience_level', 'job_title', 'salary_in_usd', 'remote_ratio', 'company_location']


* Saving processed DataFrame to a new CSV file

In [ ]:
df.to_csv('dataset/processed/salaries_processed.csv', index=False)

## Conclusion and next steps
* We have successfully extracted, transformed, and loaded the data into a new CSV file ready for data visualisation.